contracts/Network/Network.cpp

```diff
@@ -81,18 +87,17 @@ ACTION Network::listpairres(name reserve, symbol token_symbol, name token_contra
         } else {
             reservespert_table_inst.modify(itr, _self, [&](auto& s) {
                 auto res_it = find(s.reserve_contracts.begin(), s.reserve_contracts.end(), reserve);
-                if (res_it == s.reserve_contracts.end()) {
-                    s.reserve_contracts.push_back(reserve);
-                }
+                eosio_assert(res_it == s.reserve_contracts.end(), "already listed in reserve");
+                s.reserve_contracts.push_back(reserve);
             });
         }
-    } else if (token_exists) {
+    } else {
+        eosio_assert(token_exists, "not listed at all");
         bool last_reserve_for_token = false;
         reservespert_table_inst.modify(itr, _self, [&](auto& s) {
             auto res_it = find(s.reserve_contracts.begin(), s.reserve_contracts.end(), reserve);
-            if (res_it != s.reserve_contracts.end()) {
-                s.reserve_contracts.erase(res_it);
-            }
+            eosio_assert(res_it != s.reserve_contracts.end(), "not listed in reserve");
+            s.reserve_contracts.erase(res_it);
             if (!s.reserve_contracts.size()) {
                 last_reserve_for_token = true;
```

Comment:
this is not related to the fix, but why don't you erase the itr here? is it because you can't/don't want to modify the table inside the callback?
also just to make sure, these lambda functions are not async right?

Reply:
not good to erase the itr inside a call back.
lambda is sync.